docs/all-api-reference.md

```diff
@@ -3364,6 +3367,164 @@ _Appears in:_
 - [WatchNamespaces](#watchnamespaces)
 
 
+## gateway-operator.konghq.com/v2alpha1
+
+Package v2alpha1 contains API Schema definitions for the gateway-operator.konghq.com v2alpha1 API group.
+
+Package v2alpha1 contains API Schema definitions for the gateway-operator.konghq.com v2alpha1 API group
+
+- [ControlPlane](#controlplane)
+### ControlPlane
+
+
+ControlPlane is the Schema for the controlplanes API
+
+<!-- control_plane description placeholder -->
+
+| Field | Description |
+| --- | --- |
+| `apiVersion` _string_ | `gateway-operator.konghq.com/v2alpha1`
+| `kind` _string_ | `ControlPlane`
+| `metadata` _[ObjectMeta](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.33/#objectmeta-v1-meta)_ | Refer to Kubernetes API documentation for fields of `metadata`. |
+| `spec` _[ControlPlaneSpec](#controlplanespec)_ |  |
+
+
+
+### Types
+
+In this section you will find types that the CRDs rely on.
+#### ControlPlaneAdminAPI
+
+
+ControlPlaneAdminAPI defines the configuration for the DataPlane Kong Admin API.
+
+
+
+| Field | Description |
+| --- | --- |
+| `workspace` _string_ | Workspace indicates the Kong Workspace to use for the ControlPlane. If left empty then no Kong workspace will be used. |
+
+
+_Appears in:_
+- [ControlPlaneOptions](#controlplaneoptions)
+- [ControlPlaneSpec](#controlplanespec)
+
+#### ControlPlaneController
+
+
+ControlPlaneController defines a controller state for the ControlPlane.
+It overrides the default behavior as defined in the deployed operator version.
+
+
+
+| Field | Description |
+| --- | --- |
+| `name` _string_ | Name is the name of the controller. |
+| `enabled` _boolean_ | Enabled indicates whether the controller is enabled or not. |
+
+
+_Appears in:_
+- [ControlPlaneOptions](#controlplaneoptions)
+- [ControlPlaneSpec](#controlplanespec)
+- [ControlPlaneStatus](#controlplanestatus)
+
+#### ControlPlaneDataPlaneTarget
+
+
+ControlPlaneDataPlaneTarget defines the target for the DataPlane that the ControlPlane
+is responsible for configuring.
+
+
+
+| Field | Description |
+| --- | --- |
+| `type` _[ControlPlaneDataPlaneTargetType](#controlplanedataplanetargettype)_ | Type indicates the type of the DataPlane target. |
+| `url` _string_ | URL is the URL of the DataPlane target. This is used for configuring externally managed DataPlanes like those installed independently with Helm. |
+| `name` _string_ | Name is the name of the DataPlane to configure. |
+
```

Comment:
what do you think about adding as spec here the `--kong-admin-*` KIC parameters?

Reply:
This will automatically populate when we define the external dataplane spec field: https://github.com/Kong/kubernetes-configuration/pull/441/files#diff-615b50895f2f01d1d99e673686c9c067a9329d301f9a14af438b5a890574a15bR171-R182.

I've intentionally left this incomplete to specify all the other options in a smaller PR, more focused on tackling https://github.com/Kong/gateway-operator/issues/1366.

Does that sound good to you @sandromodarelli?

Reply:
that's perfect 🚀